state/cleanup.go

```diff
@@ -1065,28 +1066,28 @@ func (st *State) cleanupForceDestroyedUnit(ctx context.Context, store objectstor
 				logger.Warningf("operational errors cleaning up force destroyed unit %v in relation %v: %v", unit, relation, errs)
 			}
 			if err != nil {
-				logger.Warningf("unit %q couldn't leave scope of relation %q: %v", unitId, relation, err)
+				logger.Warningf("unit %q couldn't leave scope of relation %q: %v", unitName, relation, err)
 			}
 		}
 	} else {
-		logger.Warningf("couldn't get in-scope relations for unit %q: %v", unitId, err)
+		logger.Warningf("couldn't get in-scope relations for unit %q: %v", unitName, err)
 	}
 
 	// Detach all storage.
 	err = st.forceRemoveUnitStorageAttachments(unit)
 	if err != nil {
-		logger.Warningf("couldn't remove storage attachments for %q: %v", unitId, err)
+		logger.Warningf("couldn't remove storage attachments for %q: %v", unitName, err)
 	}
 
 	// Mark the unit dead.
-	err = applicationService.EnsureUnitDead(ctx, unitId, noopLeadershipRevoker{})
+	err = applicationService.EnsureUnitDead(ctx, coreunit.Name(unitName), noopLeadershipRevoker{})
```

Comment:
**todo:** If it is safe to not une NewName here (and validate), add a comment that state it.